### ISSUE TITLE
is it possible restore the progress of the training?

### ISSUE THREAD

**Issue by jy1989** [deepfakes/faceswap#326]
Hello 
I use parameter -s to save the model every 5 iterations.
and if I stop the training after 12 iterations. the model will store 10 iterations result.

Is there any way to restore the progress of trainning when I run `train` next time?

or if I add some new training images, is it need to run the training from the beginning?

Thank you.

**Comment by torzdf** [comment]
I wouldn't save every 5 iterations, that seems way too low. It defaults to every 100 iterations, and that should be fine.

If you stop training it saves it's current progress. Training will always resume from the last saved point (barring corruption).

**Comment by reconnais** [comment]
I believe if you stop via CTRL+C and give it sufficient time to save, it'll save even those in-between iterations.

**Comment by kvrooman** [comment]
working on a model change that I'll likely post later today or tomorrow where everything is ported to use fit_generator. Its not strictly required but it allows you to easily use model.save instead of model.save_weights. 

 This combined with load_model ( instead of load_weights ) allows you to save the model architecture, the loss history, the optimizer architecture, and the current optimizer state. Saving the optimizer state allows you to restore training progress exactly as you left off.

Also one of the nice things that fit_generator allows is callbacks which have dynamic saving, ie. only save if the training loss or any other metric has improved. You can also save much more frequently as the callback is handled by a process while the gpu continues training uniterrupted.

**Comment by bryanlyon** [comment]
Model.save is unfortunately not compatible with the multi GPU model we're using.  Please make sure that you keep a check for this in your implementation.

**Comment by kvrooman** [comment]
Hmm, there seem to be some workarounds, but I only have a single GPU so it will be hard to test for me. Thanks for the heads-up.

https://github.com/keras-team/keras/issues/8123

https://stackoverflow.com/questions/47210811/can-not-save-model-using-model-save-following-multi-gpu-model-in-keras?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa

**Comment by torzdf** [comment]
Unless I'm mistaken, this is going off topic, so closing.
